### DISCUSSION TITLE
When WatsonTcpClient is connected to server the ConnectionEventArgs.Client is null

### DISCUSSION THREAD

**Discussion by kWoren** [jchristn/WatsonTcp#240] [Q&A] [answered]
Hi,

Thank you for this great library, I have used it for some time and it made things so much easier!
In the latest release I noticed the following  problem:

var client = new WatsonTcpClient(ServerIp, ServerPort);

client.Events.ServerConnected += (s, e) =>
{
    e.Client is null here!!
};

**Comment by jchristn** [comment, marked as answer]
Hi, that is expected.  The client property is only useful in that event on the server-side.
